### PR TITLE
Simplified RuleSet implementations

### PR DESCRIPTION
**Pull request by roxspring**
- Simplified most `RuleSet` implementations by using default logic in `AbstractRuleSet` (as requested by @tkrop)
- Removed `TestRuleSet` which was accidentally left behind in refactoring
- Removed `RuleSet.title` property since it wasn't actually being used anywhere 👈 might be preferable to "fix" this by actually using the title somewhere but I figure we can add the title back if/when that comes up.

#560

### CODE REVIEW

File: server/src/test/java/de/zalando/zally/apireview/RestApiTestConfiguration.kt [hunk 1]
```diff
@@ -29,18 +28,7 @@ class RestApiTestConfiguration {
             TestCheckAlwaysReport3MustViolations()
         ))
 
-    class TestRuleSet : AbstractRuleSet() {
-
-        override val id: String = javaClass.simpleName
-
-        override val title: String = "Test Rules"
```
Comment (tkrop) [inline server/src/test/java/de/zalando/zally/apireview/RestApiTestConfiguration.kt]: Funny. Have we really nowhere use the title?

**Comment by tkrop** [comment]
:+1:

**Comment by zeitlinger** [comment]
:+1:

File: server/src/test/java/de/zalando/zally/apireview/RestApiTestConfiguration.kt (hunk 1 above)
Comment (roxspring) [reply server/src/test/java/de/zalando/zally/apireview/RestApiTestConfiguration.kt]: Really looks like we haven't. 🤦‍♂

**Pull request merged**